### ISSUE TITLE
Create Dataset: Add create global id logic to create dataset command, affects API.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#766]
Tested on 7/3

Dataset API does not generate a persistent global id because the logic to do so currently lives in the UI rather than in the create command. Needs to be added to the create command.

**Comment by scolapasta** [comment]
@ekraffmiller @sekmiller can we discuss and figure out what is needed here?

**Comment by ekraffmiller** [comment]
This is done
On Jan 23, 2015, at 2:20 PM, Gustavo Durand <notifications@github.com<mailto:notifications@github.com>> wrote:


@ekraffmiller<https://github.com/ekraffmiller> @sekmiller<https://github.com/sekmiller> can we discuss and figure out what is needed here?

**Comment by kcondon** [comment]
Create now uses DOI generation logic but I had thought it also needed to support specifying the DOI for cases where it preexisted. The test I ran was using the dataset-create-new.json file under /data:

curl -H "Content-type:application/json" -X POST -d @data/dataset-create-new.json "https://dvn-build.hmdc.harvard.edu/api/dvs/1/datasets/?key=

This worked fine -it create a dataset with a newly generated DOI. The issue is that the json actually contained a preexisting DOI so it basically ignored it. 

As a simple test, I tried removing the pesristentURL  line from the json to see whether it still worked but failed with a json format error: token=EOF. I ran it through json format checker and it was OK.

The main issues are:
1. How specify DOI using this function
2. What json format is needed for specifying doi and also for allowing the system to generate it. Currently it seems the system just ignores what is specifies but fails if any of the ignored id is removed.
